### PR TITLE
ui/temp-sched: fix out of bounds item edge cases

### LINKED ISSUE #1948: ui/temp-sched: erroneous out-of-bounds shift list item
**Describe the Bug:**
It is possible to achieve the state described below

**Steps to Reproduce:**
1. Go to temporary schedule dialog
2. configure as in screenshot below
3. see incorrect rendering of out of bounds item

**Expected Behavior:**
No out of bounds notice

**Observed Behavior:**
Out of bounds notice

**Screenshots/Stack Traces:**
Screen Shot 2021-10-04 at 8 58 25 PM

**Application Version:**
`14b9e47ed429b68fd704f82a5b31126cfe6a8e18`

**Additional Context:**
I've reproduced a failing test case on [bug-repro-out-of-bounds](https://github.com/target/goalert/compare/bug-repro-out-of-bounds). A PR to fix should include this test.

### PR DESCRIPTION
**Pull request by dctalbot**
- [x] Identified the issue which this PR solves.
- [x] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [x] Code builds clean without any errors or warnings.
- [x] Added appropriate tests for any new functionality.
- [x] All new and existing tests passed.
- [x] Added comments in the code, where necessary.
- [x] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**
This PR fixes a couple of edge cases with rendering "out of bounds" items in the shifts list (see #1948)

**Which issue(s) this PR fixes:**
Fixes #1948 

**Additional Info:**
This PR includes some refactoring as well:
- short circuit to render loading spinner (saves a lot of needless work, simplifies zone checks)
- I learned that `DateTime.min` and `DateTime.max` can take X arguments, so we leverage that
- Move `ExplicitZone` to `luxon-helpers` module

### CODE REVIEW

**Review by Forfold — APPROVED** [review]

File: web/src/app/schedules/temp-sched/shiftsListUtil.ts [hunk 1]
```diff
@@ -52,23 +58,33 @@ export function getSubheaderItems(
 export function getOutOfBoundsItems(
   schedInterval: Interval,
   shifts: Shift[],
-  zone: string,
+  zone: ExplicitZone,
 ): Sortable<FlatListNotice>[] {
-  // timezone is loaded in after initial render
-  if (zone === '') {
+  if (!schedInterval.isValid) {
```
[the remark this answers is not in the document]
Comment (dctalbot) [reply web/src/app/schedules/temp-sched/shiftsListUtil.ts:63]: Yes!
